src/coprocessor/Client.cc

```diff
@@ -525,50 +526,114 @@ std::vector<CopTask> ResponseIter::handleTaskImpl(kv::Backoffer & bo, const CopT
+        if (req_ctx->no_resp)
+            return {};
     }
     catch (Exception & e)
     {
-        bo.backoff(kv::boRegionMiss, e);
-        return buildCopTasks(bo, cluster, task.ranges, task.req, task.store_type, task.keyspace_id, log, task.meta_data, task.before_send);
-    }
-    if (resp->has_locked())
-    {
-        kv::LockPtr lock = std::make_shared<kv::Lock>(resp->locked());
-        log->debug("encounter lock problem: " + resp->locked().DebugString());
-        std::vector<uint64_t> pushed;
-        std::vector<kv::LockPtr> locks{lock};
-        auto before_expired = cluster->lock_resolver->resolveLocks(bo, task.req->start_ts, locks, pushed);
-        if (!pushed.empty())
-        {
-            min_commit_ts_pushed.addTimestamps(pushed);
-        }
-        if (before_expired > 0)
+        if (e.code() == GRPCNotImplemented)
         {
-            log->information("get lock and sleep for a while, sleep time is " + std::to_string(before_expired) + "ms.");
-            bo.backoffWithMaxSleep(kv::boTxnLockFast, before_expired, Exception(resp->locked().DebugString(), ErrorCodes::LockError));
+            // Fallback to use coprocessor rpc.
+            log->information("coprocessor stream is not implemented, fallback to use coprocessor");
+            return handle_cop_req();
         }
+        bo.backoff(kv::boRegionMiss, e);
         return buildCopTasks(bo, cluster, task.ranges, task.req, task.store_type, task.keyspace_id, log, task.meta_data, task.before_send);
     }
 
-    const std::string & err_msg = resp->other_error();
-    if (!err_msg.empty())
+    auto ret = handle_resp(std::move(resp));
+    if (!ret.empty())
+        return ret;
+
+    while (!cancelled)
     {
-        throw Exception("Coprocessor error: " + err_msg, ErrorCodes::CoprocessorError);
+        resp = std::make_shared<::coprocessor::Response>();
+        if (!req_ctx->reader->Read(resp.get()))
+            break;
+        if (resp->has_region_error())
+            throw Exception("Coprocessor stream subsequent response has region error: " + resp->region_error().message(), ErrorCodes::CoprocessorError);
+
+        /// Throw exception for the lock error from a subsequent response is ok for tiflash.
+        /// Because tiflash will resolve all locks from a region before sending a response.
+        /// However, it's not true for tikv. Currently, this logic is only used for tiflash.
+        if (resp->has_locked())
+            throw Exception("Coprocessor stream subsequent response has lock error", ErrorCodes::CoprocessorError);
+
+        const std::string & err_msg = resp->other_error();
+        if (!err_msg.empty())
+            throw Exception("Coprocessor other error: " + err_msg, ErrorCodes::CoprocessorError);
+
+        std::lock_guard<std::mutex> lk(results_mutex);
+        results.push(Result(resp));
```

Comment:
Currently all results are buffered in the memory, if the consumer cannot consume results in time. Better to be some bounded queue.